### PR TITLE
Translate TrafficPolicy for federated traffic targets

### PR DESCRIPTION
**Pull request by harveyxia**
BOT NOTES: 
resolves https://github.com/solo-io/service-mesh-hub/issues/1002
resolves https://github.com/solo-io/service-mesh-hub/issues/1045

### CODE REVIEW

File: pkg/mesh-networking/translation/utils/metautils/object_meta.go [hunk 1]
```diff
@@ -24,6 +25,28 @@ func TranslatedObjectMeta(sourceObj ezkube.ClusterResourceId, annotations map[st
...
+		ClusterName: meshInstallation.Cluster,
+		Labels:      TranslatedObjectLabels(),
+		Annotations: annotations,
+	}
+}
+
+func federatedObjectName(
+	sourceObj ezkube.ClusterResourceId,
+) string {
+	return sourceObj.GetName() + "-" + sourceObj.GetClusterName()
```
Comment (harveyxia) [inline pkg/mesh-networking/translation/utils/metautils/object_meta.go:47]: I'd like to get thoughts on this naming scheme—we need to give translation outputs for federated traffic targets a distinct name

File: pkg/mesh-networking/translation/istio/decorators/mirror/mirror_decorator.go [hunk 2]
```diff
@@ -67,9 +67,32 @@ func (d *mirrorDecorator) ApplyTrafficPolicyToVirtualService(
...
+		return err
+	}
+	if mirror != nil {
+		if err := registerField(&output.Mirror, mirror); err != nil {
+			return err
+		}
+		output.Mirror = mirror
+		output.MirrorPercentage = percentage
+	}
+	return nil
```
Comment (ilackarms) [inline pkg/mesh-networking/translation/istio/decorators/mirror/mirror_decorator.go:88]: nit: these lines are duplicated with 56-67

i'd suggest extracting these to a shared `applyTrafficPolicy` function

File: pkg/mesh-networking/translation/istio/decorators/mirror/mirror_decorator.go [hunk 3]
```diff
@@ -113,9 +138,15 @@ func (d *mirrorDecorator) makeKubeDestinationMirror(
 	mirrorKubeService := mirrorService.Spec.GetKubeService()
 
 	// TODO(ilackarms): support other types of TrafficTarget destinations, e.g. via ServiceEntries
-	localCluster := originalService.Spec.GetKubeService().GetRef().GetClusterName()
+
+	var sourceCluster string
```
Comment (ilackarms) [inline pkg/mesh-networking/translation/istio/decorators/mirror/mirror_decorator.go:142]: i'd suggest adding a comment here explaining how we derive / use `sourceCluster` here

File: pkg/mesh-networking/translation/istio/traffictarget/virtualservice/virtual_service_translator.go [hunk 4]
```diff
@@ -35,6 +36,14 @@ type Translator interface {
 		trafficTarget *discoveryv1alpha2.TrafficTarget,
 		reporter reporting.Reporter,
 	) *networkingv1alpha3.VirtualService
+
+	// TranslateFederated translates a VirtualService for a federated traffic target on the specified cluster, using global FQDNs for hostnames.
+	TranslateFederated(
```
Comment (ilackarms) [inline pkg/mesh-networking/translation/istio/traffictarget/virtualservice/virtual_service_translator.go:41]: seeing it written out makes me think we should rename `Translate` to `TranslateLocal` (and update the comment accordingly), wdyt?

File: pkg/mesh-networking/translation/utils/metautils/object_meta.go (hunk 1 above)
Comment (harveyxia) [reply pkg/mesh-networking/translation/utils/metautils/object_meta.go]: updated, see later commits

**Review by ilackarms — APPROVED** [review]

**Pull request merged**